doc/proposals/openshift-4.3/operator-testing-tool.md

```diff
@@ -0,0 +1,240 @@
+
+## Motivation
+
+Operator Testing and Validation remains difficult today due to tooling sprawl, disparate definitions of validity, and the lack of means to provide custom tests per Operator in a way that is not tied to a particular CI pipeline.
+
+Having high quality Operators is crucial to the success of the Operator Framework.
+
+### Goals
+
+ * Operator Developers can use a tool that reports on recommended, required and optional packaging requirements of a given bundle on disk
+ * Operator Developers can use the same tool that reports on recommended, required and optional common black box tests of a deployed operator on cluster
+ * Operator Developers can rely on a central source of truth of a valid Operator bundles shared among all Operator Framework components
+
+
+### Non-Goals
+
+not in scope: Operator Developers can use the same tool to run custom, functional tests of their Operator on cluster
+
+## Proposal
+
+### User Stories 
+
+#### Story 1 - Show pass/fail in Scorecard Output
+Today, the scorecard output shows a percentage of tests that were successful to the end user. This story is to change the scorecard output to show a pass or fail for each test that is run in the output instead of a success percentage. 
+
+The exit code of scorecard would be 0 if all tests passed. The exit code would be non-zero if tests failed. With this change scores are essentially replaced with a list of pass/fail(s).
+
+The scorecard would by default run all tests regardless if a single test fails.  Using a CLI flag such as below would cause the test execution to stop on a failure:
+ * operator-sdk scorecard -l ‘testgroup=required’ --fail-fast
+
+Tasks for this story include:
+ * change scorecard by removing earnedPoints and maximumPoints from each test
+ * change scorecard output by removing totalScorePercent, partialPass, earnedPoints, maximumPoints from scorecard output
+ * change scorecard existing tests to remove partialPass, instead only returning a pass or fail
+ * change scorecard CLI to return an exit code based on the pass/fail results from the set of executed tests 
+ * (stretch goal) add support for --fast-fail CLI flag
+
+
+Rough point estimate is 8 points.
+
+
+#### Story 2 - Change Scorecard Test Selection
+Today, the scorecard lets you select which tests are run by the plugins defined in the scorecard configuration (basic, olm). This story would change how users determine which tests are executed by scorecard.
+
+This story would introduce labels for each test and then allow the scorecard CLI to filter which tests it runs based on label selectors.
+
+Tests can fall into 3 groups: required, recommended, or optional. Tests can also be categorized as static or runtime. Labels for these groups and categories would allow a test to be more precisely specified by an end user.
+
+Possible examples of specifying what tests to run are as follows: 
+ * operator-sdk scorecard -l ‘testgroup in (required,recommended,optional)’ 
```

Comment:
In the practice what will be the diff between recommended and optional? In my POV that all NO mandatory recommendations are Optional. WDYT?
IMHO: It should be mandatory and recommended.

Reply:
If the optional are related to the tests impl by the devs when they extending its API and creates their plugins I think we could give a name more intuitive as plugins, or additional

Reply:
I agree. @dmesser can you describe what the difference is between recommended and optional?

As a user, why would I want to run optional, but not recommended, tests? 

Do we have a breakdown of which of the current tests fall into these two/three catagories?

If not, I think we should propose what we think and circulate with the OLM and CVP teams for feedback.